fsspec/spec.py

```diff
@@ -1437,14 +1437,28 @@ def to_dict(self) -> Dict[str, Any]:
         multiple), ``args`` (positional args, usually empty), and all other
         keyword arguments as their own keys.
         """
+        from .json import FilesystemJSONEncoder
+
+        json_encoder = FilesystemJSONEncoder()
+
+        def serialize(obj: Any) -> Any:
```

Comment:
Can we put these functions (serialize/deserialize) into the `json` module? They don't appear to depend on `self`. I'm not totally sure why this was necessary at all, I thought the whole purpose of the Encoder is to use it to walk the object tree,

Reply:
The dict methods do not use the JSON encoder/decoder since those are only capable of converting to/from JSON strings directly